pure_interface/interface.py

```diff
@@ -490,7 +477,9 @@ def _get_adapter(cls: AnInterfaceType, obj_type: Type) -> Optional[Callable]:
     """ Returns a callable that adapts objects of type obj_type to this interface or None if no adapter exists.
     """
     adapters = {}  # type: ignore
-    candidate_interfaces = [cls] + cls.__subclasses__()
+    # registered interfaces can come from cls.register(AnotherInterface) or @sub_interface_of(AnotherInterface)(cls)
+    registered_interfaces = [iface for iface in cls._pi.registered_types if type_is_interface(iface)]
```

Comment:
Isn't the `if type_is_interface(iface)` double-handling since the `register()` method checks that before adding it to `cls._pi.registered_types` set?